src/Form/FormAutosuggest.jsx

```diff
@@ -0,0 +1,295 @@
+      }));
+
+      setIsOpen(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', handleClickOutside, true);
+
+    return () => {
+      document.removeEventListener('click', handleClickOutside, true);
+    };
+  });
+
+  const setValue = (itemValue) => {
+    if (value === itemValue) { return; }
+
+    if (handleChange) { handleChange(itemValue); }
+
+    const opt = props.options.find((o) => o === itemValue);
+
+    if (opt && opt !== state.displayValue) {
+      setState(prevState => ({
+        ...prevState,
+        displayValue: opt,
+      }));
+    }
+  };
+
+  const handleItemClick = (e) => {
+    setValue(e.target.value);
+
+    setState(prevState => ({
+      ...prevState,
+      dropDownItems: '',
+    }));
+
+    setIsOpen(false);
+  };
+
+  function getItems(strToFind = '') {
+    let { options } = props;
+
+    if (strToFind.length > 0) {
+      options = options.filter((option) => (option.toLowerCase().includes(strToFind.toLowerCase())));
+    }
+
+    return options.map((opt) => {
+      let optValue = opt;
+      if (optValue.length > 30) {
+        optValue = optValue.substring(0, 30).concat('...');
+      }
+
+      return (
+        <button
+          type="button"
+          className="dropdown-item data-hj-suppress"
+          value={optValue}
+          key={optValue}
+          onClick={(e) => { handleItemClick(e); }}
+        >
+          {optValue}
+        </button>
+      );
+    });
+  }
+
+  const setDisplayValue = (itemValue) => {
+    const normalized = itemValue.toLowerCase();
+    const opt = props.options.find((o) => o.toLowerCase() === normalized);
+
+    if (opt) {
+      setValue(opt);
+      setState(prevState => ({
+        ...prevState,
+        displayValue: opt,
+      }));
+    } else {
+      setValue(null);
+      setState(prevState => ({
+        ...prevState,
+        displayValue: itemValue,
+      }));
+    }
+  };
+
+  const handleClick = (e) => {
+    const dropDownItems = getItems(e.target.value);
+
+    if (dropDownItems.length > 1) {
+      setState(prevState => ({
+        ...prevState,
+        dropDownItems,
+        errorMessage: '',
+      }));
+
+      setIsOpen(true);
+    }
+
+    if (state.dropDownItems.length > 0) {
+      setState(prevState => ({
+        ...prevState,
+        dropDownItems: '',
+        errorMessage: '',
+      }));
+
+      setIsOpen(false);
+    }
+  };
+
+  const handleOnChange = (e) => {
+    const findstr = e.target.value;
+
+    if (findstr.length) {
+      const filteredItems = getItems(findstr);
+      setState(prevState => ({
+        ...prevState,
+        dropDownItems: filteredItems,
+        errorMessage: '',
+      }));
+
+      setIsOpen(true);
+    } else {
+      setState(prevState => ({
+        ...prevState,
+        dropDownItems: '',
+        errorMessage,
+      }));
+
+      setIsOpen(false);
+    }
+
+    setDisplayValue(e.target.value);
+  };
+
+  if (handleFocus) { handleFocus(); }
```

Comment:
Why these `if (handleFocus) { handleFocus(); }` and `if (handleBlur) { handleBlur(); }` are used here?

Reply:
Deleted, thanks